cdc/scheduler/util/table_set.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"github.com/pingcap/ticdc/cdc/model"
+)
+
+// TableSet provides a data structure to store the tables' states for the
+// scheduler.
+type TableSet struct {
+	tableIDMap   map[model.TableID]*TableRecord
+	captureIndex map[model.CaptureID]map[model.TableID]*TableRecord
+}
+
+// TableRecord is a record to be inserted into TableSet.
+type TableRecord struct {
+	TableID   model.TableID
+	CaptureID model.CaptureID
+	Status    TableStatus
+}
+
+// TableStatus is a type representing the table's replication status.
+// The meaning of its value can be defined by the user of TableSet.
+type TableStatus int32
+
+// NewTableSet creates a new TableSet.
+func NewTableSet() *TableSet {
+	return &TableSet{
+		tableIDMap:   map[model.TableID]*TableRecord{},
+		captureIndex: map[model.CaptureID]map[model.TableID]*TableRecord{},
+	}
+}
+
+// AddTableRecord inserts a new TableRecord.
+// It returns true if it succeeds. Returns false if there is a duplicate.
+func (s *TableSet) AddTableRecord(record *TableRecord) (successful bool) {
```

Comment:
Does it necessary to check whether success or not?

Reply:
If there is a duplicate, usually there is a fatal bug. We would like to detect it so that the program can fail fast instead of introducing a correctness problem silently.